driver-core/src/test/unit/com/mongodb/AuthConnectionStringTest.java

```diff
@@ -93,26 +92,14 @@ private void testValidUris() {
         }
 
         MongoCredential credential = connectionString.getCredential();
-        assertString("auth.db", credential.getSource());
-        assertString("auth.username", credential.getUserName());
-
-        // Passwords for certain auth mechanisms are ignored.
-        String password = credential.getPassword() != null ? new String(credential.getPassword()) : null;
-        if (password != null) {
-            assertString("auth.password", password);
-        }
-        if (definition.get("options").isDocument()) {
-            for (Map.Entry<String, BsonValue> option : definition.getDocument("options").entrySet()) {
-                if (option.getKey().equals("authmechanism")) {
-                    String expected = option.getValue().asString().getValue();
-                    if (expected.equals("MONGODB-CR")) {
-                        assertNotNull(connectionString.getCredential());
-                        assertNull(connectionString.getCredential().getAuthenticationMechanism());
-                    } else {
-                        String actual = connectionString.getCredential().getAuthenticationMechanism().getMechanismName();
-                        assertEquals(expected, actual);
-                    }
-                }
+        if (credential != null) {
```

Comment:
Shouldn't we be asserting the expected values of the other MongoCredential fields, like `source` and `mechanismProperties`?

Reply:
`source` is already checked. I have added checks for `mechanism` and `mechanismProperties`.